src/components/LocationAutocomplete/LocationAutocompleteRenderer/LocationAutocompleteRenderer.js

```diff
@@ -0,0 +1,136 @@
+        } else {
+            setIsLoading(false);
+            resetSuggestionsList();
+        }
+    };
+
+    // eslint-disable-next-line react/display-name
+    const renderListPoweredByGoogle = ({ listInputValue, getItemProps, highlightedIndex }) => (
+        <React.Fragment>
+            <img
+                {...elem('poweredByGoogleImage', props)}
+                src={POWERED_BY_GOOGLE_ON_WHITE}
+                alt="Powered by Google"
+            />
+            {suggestionsList.map((item, index) => (
+                <ListItem
+                    key={suggestionToString(item)}
+                    {...getItemProps({
+                        item,
+                        index,
+                        isHighlighted: highlightedIndex === index,
+                        highlightContext: 'brand',
+                    })}
+                >
+                    <MarkedText marker={listInputValue} inline>
+                        {suggestionToString(item)}
+                    </MarkedText>
+                </ListItem>
+            ))}
+        </React.Fragment>
+    );
+
+    return (
+        <Autosuggest
+            getSuggestions={suggestionsList}
+            suggestionToString={suggestionToString}
+            isLoading={isLoading}
+            inputPlaceholder={inputPlaceholder}
+            noSuggestionsPlaceholder={noSuggestionsPlaceholder}
+            listRenderer={renderListPoweredByGoogle}
+            onBlur={resetSuggestionsList}
+            onInputValueChange={handleInputValueChange}
+            onSelectionChange={onSelectionChange}
+            {...rest}
+        />
+    );
+};
+
+LocationAutocomplete.displayName = 'LocationAutocompleteRenderer';
+
+LocationAutocomplete.propTypes = {
+    /** to be shown in the input field when no value is typed */
+    inputPlaceholder: PropTypes.string.isRequired,
+    /** to be shown when no suggestions are available */
+    noSuggestionsPlaceholder: PropTypes.string.isRequired,
+    /** callback to be called with selected value.
+     * Value is of type [AutocompletePrediction](https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletePrediction)
+     */
+    onSelectionChange: PropTypes.func,
+    /** restrict predictions to country/countries.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#ComponentRestrictions
+     */
+    country: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
+    /** type of locations that should be searched for.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletionRequest.types
+     */
+    placeTypes: PropTypes.arrayOf(PropTypes.string),
+};
+
+LocationAutocomplete.defaultProps = {
+    onSelectionChange: () => null,
+    country: null,
+    placeTypes: ['(regions)'],
+};
+
+export default LocationAutocomplete;
```

Comment:
I don't see the reason to use separate component **LocationAutocompleteRenderer**. We can implement this logic in **LocationAutocomplete**. Now LocationAutocompleteRenderer component is very coupled with LocationAutocomplete, and we can't reuse it. Or to implement all google logic (init/retrieve suggestions, etc) in one component LocationAutocomplete and then LocationAutocompleteRenderer will not aware about google, just takes care about render suggestions, debounce, clear, etc.

Reply:
There are 2 reasons for this split: 
1. to make it testable
2. the wrapper that loads google can be easily extracted. So in LocationSelector, we can have one loader that loads all APIs at once

we cannot extract google completely as it's the main point of this component - managing info from google. The rest of the logic is already extracted to Autosuggest